src/Workleap.DomainEventPropagation.Subscription.PullDelivery/EventPropagationSubscriberBuilder.cs

```diff
@@ -11,11 +13,18 @@ namespace Workleap.DomainEventPropagation;
 
 internal sealed class EventPropagationSubscriberBuilder : IEventPropagationSubscriberBuilder
 {
+    private readonly DomainEventTypeRegistry _domainEventTypeRegistry;
+
     public EventPropagationSubscriberBuilder(IServiceCollection services)
     {
         this.Services = services;
 
-        this.Services.AddSingleton<IDomainEventTypeRegistry, DomainEventTypeRegistry>();
+        this._domainEventTypeRegistry = services.Where(x => x.ServiceType == typeof(IDomainEventTypeRegistry))
+            .Select(x => x.ImplementationInstance)
+            .OfType<DomainEventTypeRegistry>()
+            .FirstOrDefault() ?? new DomainEventTypeRegistry();
```

Comment:
What would be the use case here to register many `IDomainEventTypeRegistry` implementations?

Reply:
I guess this could happen if you register this lib muliple times ? Not entirely sure (I copied this bit from what was done in the existing lib)

Reply:
If you use `TryAddSingleton`, then multiple calls to the registration method shall register it once.

Reply:
L26 the service is already added as a singleton (TryAdd). We only do this to either retrieve the existing registry, or we new it if it doesn't exist